library/src/main/java/com/qiniu/android/http/request/HttpSingleRequest.java

```diff
@@ -149,29 +155,46 @@ private synchronized void completeAction(Request request,
                                              UploadSingleRequestMetrics requestMetrics,
                                              RequestCompleteHandler completeHandler) {
 
-        if (client == null){
+        if (client == null) {
             return;
         }
         client = null;
 
-        if (completeHandler != null){
+        updateHostNetworkStatus(responseInfo, request.uploadServer, requestMetrics);
+        reportRequest(responseInfo, request, requestMetrics);
+
+        if (completeHandler != null) {
             completeHandler.complete(responseInfo, requestMetricsList, response);
         }
-        reportRequest(responseInfo, request, requestMetrics);
+    }
+
+    private void updateHostNetworkStatus(ResponseInfo responseInfo, IUploadServer server, UploadSingleRequestMetrics requestMetrics) {
+        if (requestMetrics == null) {
+            return;
+        }
+        long byteCount = requestMetrics.bytesSend();
+        if (requestMetrics.startDate != null && requestMetrics.endDate != null && byteCount > 1024 * 1024) {
+            double second = requestMetrics.endDate.getTime() - requestMetrics.endDate.getTime();
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Nonsensical self computation involving a value (e.g., x & x) (SA_SELF_COMPUTATION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-nonsensical-self-computation-involving-a-field-e-g-x-x-sa-field-self-computation).
This method performs a nonsensical computation of a field with another reference to the same field (e.g., x&x or x-x). Because of the nature of the computation, this operation doesn't seem to make sense, and may indicate a typo or a logic error. Double check the computation.